flask_oidc/__init__.py

```diff
@@ -119,32 +122,45 @@ def __init__(self, app=None, credentials_store=None, http=None, time=None,
         # By default, we do not have a custom callback
         self._custom_callback = None
 
+        # In the beginning we know nothing ...
+        self.client_secrets = None
+
         # get stuff from the app's config, which may override stuff set above
-        if app is not None:
+        
+        if app:
             self.init_app(app)
 
+        # Backwards compatible: When provider argument is ommitted,
+        # the value would be None, then in method load_secrets, the
+        # client_secrets will be tried to load...
+        with app.app_context():
```

Comment:
This would need to move to `init_app`, since when we get here, the developer might not yet have provided an `app` instance.